src/test/java/com/github/neatlife/jframework/util/PrecisionUtilTest.java

```diff
@@ -0,0 +1,137 @@
+package com.github.neatlife.jframework.util;
+
+import org.junit.Assert;
+import org.junit.Test;
+import java.math.BigDecimal;
+
+public class PrecisionUtilTest {
+
+    @Test
+    public void testBigDecimalAdd() {
+        Assert.assertEquals(BigDecimal.ONE,
+                PrecisionUtil.add(null, BigDecimal.ONE));
+        Assert.assertEquals(BigDecimal.ONE,
+                PrecisionUtil.add(BigDecimal.ONE, null));
+    }
+
+    @Test
+    public void testBigDecimalSubtract() {
+        Assert.assertEquals(new BigDecimal("-10"),
+                PrecisionUtil.subtract(null, BigDecimal.TEN));
+        Assert.assertEquals(BigDecimal.ONE,
+                PrecisionUtil.subtract(BigDecimal.ONE, null));
+    }
+
+    @Test
+    public void testBigDecimalMultiply() {
+        Assert.assertEquals(new BigDecimal("10"),
+                PrecisionUtil.multiply(null, BigDecimal.TEN));
+        Assert.assertEquals(BigDecimal.ONE,
+                PrecisionUtil.multiply(BigDecimal.ONE, null));
+    }
+
+    @Test
+    public void testBigDecimalDivide() {
+        Assert.assertEquals(new BigDecimal("0.10"),
+                PrecisionUtil.divide(BigDecimal.ONE, BigDecimal.TEN));
+        Assert.assertEquals(new BigDecimal("0.10"),
+                PrecisionUtil.divide(BigDecimal.ONE, BigDecimal.TEN));
+        Assert.assertEquals(new BigDecimal("0"),
+                PrecisionUtil.divide(null, BigDecimal.TEN, 1));
+        Assert.assertEquals(new BigDecimal("1.0"),
+                PrecisionUtil.divide(BigDecimal.ONE, null, 1));
```

Comment:
In this test change the first param to something other than `ONE` so it's clear that the non-null param is not the source of the return value.

Reply:
Done